src/middlewares.js

```diff
@@ -84,60 +85,75 @@ function handleParseHeaders(req, res, next) {
   }
 
   info.app = cache.apps.get(info.appId);
-  req.config = new Config(info.appId, mount);
-  req.info = info;
+  let fetchConfiguration = Promise.resolve();
+  if (info.app.enableConfigChanges) {
+    let settingsManager = SettingsManager(info.appId);
+    fetchConfiguration = settingsManager.pull()
+      .then(settingsManager.updateCache);
+  }
 
-  var isMaster = (info.masterKey === req.config.masterKey);
+  return fetchConfiguration
+    .catch(error => {
+      log.error('error retrieving server settings from database', error);
+      throw new Parse.Error(Parse.Error.UNKNOWN_ERROR, error);
```

Comment:
Instead of `UNKNOWN_ERROR` let's use `INTERNAL_SERVER_ERROR`. I think that makes more sense for errors caused by the database.